File: Leetcode/Python/344. Reverse String/solution.py
# https://leetcode.com/problems/reverse-string/

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        length = len(s)
        for idx in range(length // 2):
            s[idx], s[length - idx - 1] = s[length - idx - 1], s[idx]


class Solution:
    def reverseString(self, s: List[str]) -> None:
        s.reverse()


class Solution:
    def reverseString(self, s: List[str]) -> None:
        s[:] = s[::-1]

File: Leetcode/Typescript/344. Reverse String/solution.ts
// https://leetcode.com/problems/reverse-string/description/

/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
  let left = 0,
    right = s.length - 1;

  while (left < right) {
    const temp = s[left];
    s[left] = s[right];
    s[right] = temp;

    left += 1;
    right -= 1;
  }
}
